### PR TITLE
[emergency-kill] Kill is a setting. To be less xml dependent it shoul…

### PR DESCRIPTION
**Pull request by dewagter**
…d be a message so that several boards can be killed together

### CODE REVIEW

File: sw/ground_segment/cockpit/live.ml [hunk 1]
```diff
@@ -457,6 +461,10 @@ let create_ac = fun ?(confirm_kill=true) alert (geomap:G.widget) (acs_notebook:G
   let via_http = Str.string_match (Str.regexp "http") af_url 0 in
   let af_xml = ExtXml.parse_file ~noprovedtd:via_http af_file in
 
+  (** Get firmware name *)
+  let firmware = ExtXml.child af_xml "firmware" in
```
Comment (gautierhattenberger) [inline sw/ground_segment/cockpit/live.ml]: you can have several firmwares in a xml airframe file

File: sw/ground_segment/cockpit/live.ml [hunk 2]
```diff
@@ -755,11 +763,15 @@ let create_ac = fun ?(confirm_kill=true) alert (geomap:G.widget) (acs_notebook:G
           let connect = fun ?(warning=true) setting_name strip_connect ->
             try
               let id = settings_tab#assoc setting_name in
-              strip_connect (fun x -> dl_setting_callback id x)
+              if setting_name = "kill_throttle" then
```
Comment (gautierhattenberger) [inline sw/ground_segment/cockpit/live.ml]: If kill connect is made a specific command, I would also make a dedicated `connect` function, there is no real gain here.

File: sw/ground_segment/cockpit/live.ml [hunk 3]
```diff
@@ -253,6 +253,10 @@ let dl_setting = fun ac_id idx value ->
   let vs = ["ac_id", PprzLink.String ac_id; "index", PprzLink.Int idx;"value", PprzLink.Float value] in
   Ground_Pprz.message_send "dl" "DL_SETTING" vs
 
+let dl_emergency_cmd = fun ac_id cmd ->
+  let vs = ["ac_id", PprzLink.String ac_id; "cmd", PprzLink.Int cmd] in
+  Ground_Pprz.message_send "dl" "DL_EMERGENCY_CMD" vs
```
Comment (gautierhattenberger) [inline sw/ground_segment/cockpit/live.ml]: This message doesn't exist yet in pprzlink

Comment (dewagter) [inline sw/ground_segment/cockpit/live.ml]: https://github.com/paparazzi/pprzlink/pull/34

**Comment by dewagter** [comment]
The kill button in the GCS is now sending a setting. For each board/ap/fbw/ac this setting has a different number, which is potentially dangerous. The MD5 check solves a lot of it, but in FBW/AP boards this can still go wrong when the FBW and AP both have a datalink.

In the long run I think it is better to remove the kill setting completely and replace it with a dedicate well documented datalink message. This would also make paparazzi more compatible with other ground stations (e.g. MavLink).

File: sw/ground_segment/cockpit/live.ml (hunk 2 above)
Comment (dewagter) [inline sw/ground_segment/cockpit/live.ml]: the message should be attached to the kill button of the GCS but we do not want to break kill settings

File: sw/ground_segment/cockpit/live.ml (hunk 1 above)
Comment (dewagter) [inline sw/ground_segment/cockpit/live.ml]: @fvantienen why can it not be sent to fixedwing?

File: sw/ground_segment/cockpit/live.ml (hunk 2 above)
Comment (gautierhattenberger) [inline sw/ground_segment/cockpit/live.ml]: still, I think it looks cleaner with a dedicated kill function that also send a setting if it exists

Comment (dewagter) [inline sw/ground_segment/cockpit/live.ml]: @fvantienen could you help with this?

File: sw/ground_segment/cockpit/live.ml (hunk 1 above)
Comment (dewagter) [inline sw/ground_segment/cockpit/live.ml]: Maybe this part should have been in another pull request then. Is a NO_WIND option in link also an option? Is no rotorcraft ever using wind?

**Comment by flixr** [comment]
As @gautierhattenberger already pointed out, parsing the firmware that way is not robust at all.
Disabling wind info should be in a separate PR anyway...
Also the other changes to parse message are not related to the kill setting/message...

About the kill message, I think it is a good idea. But with this change committed kill won't work anymore until people update the firmware on their aircraft, since the setting is not applied anymore.
That could be potentially dangerous, so it might be a good idea to update the setting and send the message or find some other way to make this backwards compatible.

**Comment by flixr** [comment]
Also the conflicts in pprzlink need to be fixed...

**Comment by dewagter** [comment]
- [x]  setting is still being applied
 - [x] firmware -> wind problem removed

**Comment by flixr** [comment]
LGTM
@gautierhattenberger ?

**Pull request merged**